src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -506,6 +508,61 @@ long IslandWindow::_calculateTotalSize(const bool isWidth, const long clientSize
     case WM_THEMECHANGED:
         UpdateWindowIconForActiveMetrics(_window.get());
         return 0;
+    case WM_WINDOWPOSCHANGING:
+    {
+        // GH#10274 - if the quake window gets moved to another monitor via aero
+        // snap (win+shift+arrows), then re-adjust the size for the new monitor.
+        if (IsQuakeWindow())
+        {
+            // Retrieve the suggested dimensions and make a rect and size.
+            LPWINDOWPOS lpwpos = (LPWINDOWPOS)lparam;
+
+            // We only need to apply restrictions if the position is changing.
+            // The SWP_ flags are confusing to read. This is
+            // "if we're not NOT moving the window"
+            if (WI_IsFlagClear(lpwpos->flags, SWP_NOMOVE))
+            {
+                // Figure out the suggested dimensions and position.
+                RECT rcSuggested;
+                rcSuggested.left = lpwpos->x;
+                rcSuggested.top = lpwpos->y;
+                rcSuggested.right = rcSuggested.left + lpwpos->cx;
+                rcSuggested.bottom = rcSuggested.top + lpwpos->cy;
+
+                // Find the bounds of the current monitor, and the monitor that
+                // we're suggested to be on.
+
+                RECT windowRect = GetWindowRect();
+                HMONITOR current = MonitorFromRect(&windowRect, MONITOR_DEFAULTTONEAREST);
+                MONITORINFO currentInfo;
+                currentInfo.cbSize = sizeof(MONITORINFO);
+                GetMonitorInfo(current, &currentInfo);
+
+                HMONITOR proposed = MonitorFromRect(&rcSuggested, MONITOR_DEFAULTTONEAREST);
+                MONITORINFO proposedInfo;
+                proposedInfo.cbSize = sizeof(MONITORINFO);
+                GetMonitorInfo(proposed, &proposedInfo);
+
+                // If the monitor changed...
+                if (til::rectangle{ proposedInfo.rcMonitor } !=
+                    til::rectangle{ currentInfo.rcMonitor })

```

Comment:
I don't believe you have to wrap these structs first. You should be able to compare them directly.

Reply:
Uhg, it doesn't let me do that 😕 I didn't really feel like manually comparing each of `.left`, `.top`, etc, and this isn't going to be a particularly hot path

Reply:
Oh I just wrote it because I thought it might make the code simpler. ^^